block_util/src/async_io.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright © 2021 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 AND BSD-3-Clause
+
+use vmm_sys_util::eventfd::EventFd;
+
+#[derive(Debug)]
+pub enum DiskFileError {
+    /// Failed getting disk file size.
+    Size(std::io::Error),
+    /// Failed creating a new AsyncIo.
+    NewAsyncIo(std::io::Error),
+}
+
+pub type DiskFileResult<T> = std::result::Result<T, DiskFileError>;
+
+pub trait DiskFile: Send + Sync {
+    fn size(&self) -> DiskFileResult<u64>;
+    fn new_async_io(&self, ring_depth: u32) -> DiskFileResult<Box<dyn AsyncIo>>;
+}
+
+#[derive(Debug)]
+pub enum AsyncIoError {
+    /// Failed vectored reading.
+    ReadVectored(std::io::Error),
+    /// Failed vectored writing.
+    WriteVectored(std::io::Error),
+    /// Failed synchronizing.
+    Synchronize(std::io::Error),
+}
```

Comment:
Not sure it's good that these don't complete match the names of the functions themselves.

Reply:
Ok I'll change `Synchronize` into `Fsync`.